projects/js-packages/ai-client/src/suggestions-event-source/index.ts

```diff
@@ -0,0 +1,221 @@
+		super();
+		this.fullMessage = '';
+		this.isPromptClear = false;
+
+		// The AbortController is used to close the fetchEventSource connection
+		this.controller = new AbortController();
+
+		this.initEventSource( data );
+	}
+
+	async initEventSource( {
+		url,
+		question,
+		token,
+		options = {},
+	}: SuggestionsEventSourceConstructorArgs ) {
+		const bodyData = { post_id: options?.postId, messages: [], question: '', feature: '' };
+
+		// If the url is not provided, we use the default one
+		if ( ! url ) {
+			const urlHandler = new URL( 'https://public-api.wordpress.com/wpcom/v2/jetpack-ai-query' );
+
+			// Support response from cache option
+			if ( options?.fromCache ) {
+				urlHandler.searchParams.append( 'stream_cache', 'true' );
+			}
+
+			url = urlHandler.toString();
+			debug( 'URL not provided, using default: %o', url );
+		}
+
+		// question can be a string or an array of PromptItemProps
+		if ( Array.isArray( question ) ) {
+			bodyData.messages = question;
+		} else {
+			bodyData.question = question;
+		}
+
+		// Propagate the feature option
+		if ( options?.feature?.length ) {
+			debug( 'Feature: %o', options.feature );
+			bodyData.feature = options.feature;
+		}
+
+		await fetchEventSource( url, {
+			method: 'POST',
+			headers: {
+				'Content-type': 'application/json',
+				Authorization: 'Bearer ' + token,
+			},
+			body: JSON.stringify( bodyData ),
+
+			onclose: () => {
+				debug( 'Stream closed' );
+			},
+
+			onerror: err => {
+				this.processErrorEvent( err );
+				throw err; // rethrow to stop the operation otherwise it will retry forever
+			},
+
+			onmessage: ev => {
+				this.processEvent( ev );
+			},
+
+			onopen: async response => {
+				if ( response.ok ) {
+					return;
+				}
+				if (
+					response.status >= 400 &&
+					response.status <= 500 &&
+					! [ 422, 429 ].includes( response.status )
+				) {
+					this.processConnectionError( response );
+				}
+
+				/*
+				 * error code 503
+				 * service unavailable
+				 */
+				if ( response.status === 503 ) {
+					this.dispatchEvent( new CustomEvent( 'error_service_unavailable' ) );
+				}
+
+				/*
+				 * error code 429
+				 * you exceeded your current quota please check your plan and billing details
+				 */
+				if ( response.status === 429 ) {
+					this.dispatchEvent( new CustomEvent( 'error_quota_exceeded' ) );
+				}
+
+				/*
+				 * error code 422
+				 * request flagged by moderation system
+				 */
+				if ( response.status === 422 ) {
+					this.dispatchEvent( new CustomEvent( 'error_moderation' ) );
+				}
+
+				throw new Error();
+			},
+
+			signal: this.controller.signal,
+		} );
+	}
+
+	checkForUnclearPrompt() {
+		if ( this.isPromptClear ) {
+			return;
+		}
+
+		/*
+		 * Sometimes the first token of the message is not received,
+		 * so we check only for JETPACK_AI_ERROR, ignoring:
+		 * - the double underscores (italic markdown)
+		 * - the doouble asterisks (bold markdown)
```

Comment:
```suggestion
		 * - the double asterisks (bold markdown)
```

Reply:
Thanks. Let's considering to fix it in a follow-up 🤚

Reply:
ddc4ae5e0be005f329e2e9791fbdc260560159e6